mitmproxy/connections.py

```diff
@@ -38,6 +39,7 @@ def __init__(self, client_connection, address, server):
             self.rfile = None
             self.address = None
             self.clientcert = None
+            self.mitmcert = None
```

Comment:
Your failing tests a probably because of this here.
Try moving it one indentation level out (and down to the others), then it sets this attribute always - and not only inside this if.